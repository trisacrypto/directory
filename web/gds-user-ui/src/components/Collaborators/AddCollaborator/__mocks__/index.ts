
export const collaboratorMockValue: any = {
    data: [
        {
            id: "1",
            name: "test",
            email: "test@local.dev",
            roles: ["admin"],
            created_at: "2021-01-01",
        },
    ],
};

export const userOrganizationMockValue: any = {
    data: {
        id: "1",
        name: "test",
        domain: "test.local.dev",
        created_at: "2021-01-01",
    },
};
