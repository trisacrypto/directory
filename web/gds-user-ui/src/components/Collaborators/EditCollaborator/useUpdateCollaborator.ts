import { useMutation } from '@tanstack/react-query';
import { updateCollaborator as updateCollaboratorService } from 'modules/dashboard/collaborator/CollaboratorService';
import type { UpdateCollaboratorMutation } from 'components/Collaborators/EditCollaborator/UpdateCollaboratorType';

export function useUpdateCollaborator(): UpdateCollaboratorMutation {
    const mutation = useMutation(['update-Collaborator'], updateCollaboratorService);
    return {
        updateCollaborator: mutation.mutate,
        reset: mutation.reset,
        collaborator: mutation.data,
        hasCollaboratorFailed: mutation.isError,
        wasCollaboratorUpdated: mutation.isSuccess,
        isUpdating: mutation.isLoading,
        errorMessage: mutation.error,
    };
}
