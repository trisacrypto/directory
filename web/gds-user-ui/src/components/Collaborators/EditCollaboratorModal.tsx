import {
  Box,
  Button,
  chakra,
  Link,
  Modal,
  ModalBody,
  ModalCloseButton,
  ModalContent,
  ModalFooter,
  ModalHeader,
  ModalOverlay,
  Text,
  useDisclosure,
  VStack
} from '@chakra-ui/react';
import { useState, useEffect } from 'react';
import { Trans } from '@lingui/macro';
import SelectFormControl from 'components/ui/SelectFormControl';
import { FiEdit } from 'react-icons/fi';
import { useFetchCollaborators } from 'components/Collaborators/useFetchCollaborator';
import type { Collaborator } from 'components/Collaborators/CollaboratorType';
interface Props {
  collaboratorId: string;
}
function EditCollaboratorModal({ collaboratorId }: Props) {
  const { isOpen, onOpen, onClose } = useDisclosure();
  const { collaborators } = useFetchCollaborators();
  const [collaborator, setCollaborator] = useState<Collaborator>();

  const updateHandler = () => {
    // delete collaborator
  };

  useEffect(() => {
    const col = collaborators?.data.collaborators.find(
      (c: Collaborator) => c.id === collaboratorId
    );
    if (col) {
      setCollaborator(col);
    }
  }, [collaboratorId, collaborators]);

  return (
    <Link color="blue" onClick={onOpen}>
      <FiEdit fontSize="24px" />
      <Modal isOpen={isOpen} onClose={onClose}>
        <ModalOverlay />
        <ModalContent>
          <ModalHeader textTransform="capitalize" textAlign="center" fontWeight={700}>
            <Trans>Edit collaborator Role</Trans>
          </ModalHeader>
          <ModalCloseButton />
          <ModalBody>
            <Text>
              <Trans>Select the collaboratorâ€™s role and save.</Trans>
            </Text>
            <VStack align="start" spacing={4} mt={2}>
              <Box>
                <Text fontWeight={700}>
                  <Trans>Collaborator Name & Email</Trans>
                </Text>
                <Text textTransform="capitalize">{collaborator?.name}</Text>
                <Text>{collaborator?.email}</Text>
              </Box>
              <SelectFormControl
                label={
                  <>
                    <chakra.span fontWeight={700}>
                      <Trans>Change Role</Trans>
                    </chakra.span>
                  </>
                }
                controlId="role"
              />
            </VStack>
          </ModalBody>

          <ModalFooter display="flex" flexDir="column" gap={3}>
            <Button bg="orange" minW="150px" _hover={{ bg: 'orange' }} onClick={updateHandler}>
              <Trans>Save</Trans>
            </Button>
            <Button variant="ghost" minW="150px" color="link" onClick={onClose}>
              <Trans>Close</Trans>
            </Button>
          </ModalFooter>
        </ModalContent>
      </Modal>
    </Link>
  );
}

export default EditCollaboratorModal;
