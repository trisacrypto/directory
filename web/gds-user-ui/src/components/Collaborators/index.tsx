import { Heading, Stack } from "@chakra-ui/react";
import CollaboratorsSection from "components/CollaboratorsSection";

const Collaborators: React.FC = () => {
  return (
    <Stack>
      <Heading marginBottom="69px">Collaborators</Heading>
      <CollaboratorsSection />
    </Stack>
  );
};

export default Collaborators;
