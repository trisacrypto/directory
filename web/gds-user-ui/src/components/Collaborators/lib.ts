import Store from 'application/store';

// map collaborators and set date to date object
export const mapCollaborators = (arr: any) => {
    return arr.map((collaborator: any) => {
        return {
            ...collaborator,
            created_at: new Date(collaborator.created_at),
        };
    });
};

// list by recent date
export const sortCollaboratorsByRecentDate = (arr: any) => {
    const refactoredArr = mapCollaborators(arr);
    return refactoredArr.sort((a: any, b: any) => b.created_at.getTime() - a.created_at.getTime());
};

// is collaborator current user
export const isCurrentUser = (collaboratorEmail: string): boolean => {
    return collaboratorEmail === Store.getState()?.user?.user?.email;
};


