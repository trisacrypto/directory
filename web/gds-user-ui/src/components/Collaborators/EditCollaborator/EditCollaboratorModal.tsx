import {
  Box,
  Button,
  chakra,
  Modal,
  ModalBody,
  ModalCloseButton,
  ModalContent,
  ModalFooter,
  ModalHeader,
  ModalOverlay,
  Text,
  useDisclosure,
  VStack,
  useToast
} from '@chakra-ui/react';
import { Trans } from '@lingui/react';
import SelectFormControl from 'components/ui/SelectFormControl';
import { FiEdit } from 'react-icons/fi';
import type { Collaborator } from 'components/Collaborators/CollaboratorType';
import { useUpdateCollaborator } from 'components/Collaborators/EditCollaborator/useUpdateCollaborator';
import { useFetchCollaborators } from 'components/Collaborators/useFetchCollaborator';
import React, { useEffect, useState } from 'react';
import { t } from '@lingui/macro';
import { USER_PERMISSION } from 'types/enums';
import { useSafeDisableButton } from 'components/Collaborators/useSafeDisableButton';
interface Props {
  collaboratorId: string;
  roles?: string[];
}

function EditCollaboratorModal(props: Props) {
  const { collaboratorId, roles } = props;
  const { isOpen, onOpen, onClose } = useDisclosure();
  const toast = useToast();
  const { collaborators, getAllCollaborators } = useFetchCollaborators();
  const [collaborator, setCollaborator] = useState<Collaborator>();
  const [selectedRole, setSelectedRole] = useState('');

  const {
    isUpdating,
    wasCollaboratorUpdated,
    updateCollaborator,
    hasCollaboratorFailed,
    errorMessage
  } = useUpdateCollaborator();

  const { isDisabled: isNotCurrentUserAndHasPermission } = useSafeDisableButton(
    USER_PERMISSION.UPDATE_COLLABORATOR,
    collaborator?.email as string
  );

  const updateHandler = () => {
    // update user role
    const collaboratorData = {
      data: {
        roles: new Array(selectedRole)
      },
      id: collaboratorId
    };
    updateCollaborator(collaboratorData);

    onClose();
  };

  // roles options from userRoles {label: string , value: string}
  const rolesOptions = roles?.map((v: string) => ({
    label: v,
    value: v
  }));

  // const shouldDisableButton = () => {
  //   if (isCurrentUser(collaborator?.email as string)) {
  //     return true;
  //   }
  //   return hasPermission(USER_PERMISSION.UPDATE_COLLABORATOR);
  // };

  useEffect(() => {
    let once = false;
    const col = collaborators?.find((c: Collaborator) => c.id === collaboratorId);
    if (col) {
      if (!once) {
        setCollaborator(col);
      }
    }
    return () => {
      once = true;
    };
  }, [collaboratorId, collaborators]);

  useEffect(() => {
    if (wasCollaboratorUpdated) {
      getAllCollaborators();
      // display success toast
      toast({
        title: 'Collaborator updated',
        description: 'The collaborator has been updated',
        status: 'success',
        duration: 9000,
        isClosable: true,
        position: 'top-right'
      });
    }
  }, [wasCollaboratorUpdated, getAllCollaborators, toast]);

  // if we roles is set in collaborator, set the selectedRole to the first role
  useEffect(() => {
    if (collaborator?.roles) {
      setSelectedRole(collaborator.roles[0]);
    }
  }, [collaborator]);

  useEffect(() => {
    if (hasCollaboratorFailed && !wasCollaboratorUpdated) {
      // const hasErrored =
      //   errorMessage &&
      //   t`An error occurred while updating the collaborator, please try again or contact support at support@trisa.io`;
      toast({
        title: t`Collaborator is not updated`,
        description: errorMessage || t`The collaborator has not been updated`,
        status: 'error',
        duration: 9000,
        isClosable: true,
        position: 'top-right'
      });
    }
  }, [hasCollaboratorFailed, wasCollaboratorUpdated, errorMessage, toast]);

  return (
    <Button
      color="blue"
      onClick={onOpen}
      data-testid="collaborator-button"
      bg={'transparent'}
      disabled={!isNotCurrentUserAndHasPermission}
      _hover={{
        bg: 'transparent'
      }}
      _focus={{
        bg: 'transparent'
      }}>
      <FiEdit fontSize="24px" />
      <Modal isOpen={isOpen} onClose={onClose}>
        <ModalOverlay />
        <ModalContent>
          <ModalHeader textTransform="capitalize" textAlign="center" fontWeight={700}>
            <Trans id="Edit collaborator Role">Edit collaborator Role</Trans>
          </ModalHeader>
          <ModalCloseButton />
          <ModalBody>
            <Text>
              <Trans id="Select the collaborator’s role and save.">
                Select the collaborator’s role and save.
              </Trans>
            </Text>
            <VStack align="start" spacing={4} mt={2}>
              <Box>
                <Text fontWeight={700}>
                  <Trans id="Collaborator Name & Email">Collaborator Name & Email</Trans>
                </Text>
                <Text textTransform="capitalize" data-testid="collaborator-name">
                  {collaborator?.name}
                </Text>
                <Text data-testid="collaborator-email">{collaborator?.email}</Text>
              </Box>
              <SelectFormControl
                label={
                  <>
                    <chakra.span fontWeight={700}>
                      <Trans id="Change Role">Change Role</Trans>
                    </chakra.span>
                  </>
                }
                options={rolesOptions}
                controlId="role"
                onChange={(s: any) => {
                  setSelectedRole(s.value);
                }}
                value={rolesOptions?.find((v) => v.value === selectedRole)}
                name="role"
                placeholder="Select Role"
              />
            </VStack>
          </ModalBody>

          <ModalFooter display="flex" flexDir="column" gap={3}>
            <Button
              bg="orange"
              minW="150px"
              _hover={{ bg: 'orange' }}
              data-testid="update-collaborator-button"
              onClick={updateHandler}
              isDisabled={!selectedRole}
              isLoading={isUpdating}>
              <Trans id="Save">Save</Trans>
            </Button>
            <Button
              variant="ghost"
              minW="150px"
              color="link"
              onClick={onClose}
              isLoading={isUpdating}>
              <Trans id="Close">Close</Trans>
            </Button>
          </ModalFooter>
        </ModalContent>
      </Modal>
    </Button>
  );
}

export default EditCollaboratorModal;
