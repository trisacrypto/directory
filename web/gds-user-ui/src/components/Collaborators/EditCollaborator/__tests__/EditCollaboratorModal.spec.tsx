/* eslint-disable @typescript-eslint/no-unused-vars */
import React from 'react';
import { fireEvent, screen } from '@testing-library/react';
import { useMutation } from '@tanstack/react-query';
import userEvent from '@testing-library/user-event';
import { dynamicActivate } from 'utils/i18nLoaderHelper';
import nock from 'nock';
import EditCollaboratorModal from '../EditCollaboratorModal';
import { act, render } from 'utils/test-utils';
import { collaboratorMockValue } from 'components/Collaborators/AddCollaborator/__mocks__';
import * as useCollaborators from 'components/Collaborators/useFetchCollaborator';
import * as useUpdateCollaborator from '../useUpdateCollaborator';
const mockUseMutation = useMutation as jest.Mock;

const divWithChildrenMock = (children: any, identifier: any) => (
  <div data-testId={identifier}>{children}</div>
);
const divWithoutChildrenMock = (identifier: any) => <div data-testId={identifier} />;

jest.mock('@chakra-ui/react', () => ({
  ...jest.requireActual('@chakra-ui/react'),
  Modal: jest.fn(({ children }) => divWithChildrenMock(children, 'modal')),
  ModalOverlay: jest.fn(({ children }) => divWithChildrenMock(children, 'overlay')),
  ModalContent: jest.fn(({ children }) => divWithChildrenMock(children, 'content')),
  ModalHeader: jest.fn(({ children }) =>
    divWithChildrenMock(children, 'update-collaborator-modal')
  ),
  ModalFooter: jest.fn(({ children }) => divWithChildrenMock(children, 'footer')),
  ModalBody: jest.fn(({ children }) => divWithChildrenMock(children, 'body')),
  ModalCloseButton: jest.fn(() => divWithoutChildrenMock('close'))
}));

// render delete collaborator component
function renderComponent() {
  const Props = {
    collaboratorId: '1',
    roles: ['ADMIN']
  };
  return render(<EditCollaboratorModal {...Props} />);
}

const mockCollaborators = jest.fn();
const mockGetAllCollaborators = jest.fn();
const mockUpdateCollaborator = jest.fn();
const mockCollaboratorsData = collaboratorMockValue.data;

const useFetchCollaboratorsMock = jest.spyOn(useCollaborators, 'useFetchCollaborators');
const useUpdateCollaboratorMock = jest.spyOn(useUpdateCollaborator, 'useUpdateCollaborator');
describe('UpdateCollaboratorModal', () => {
  beforeAll(() => {
    act(() => {
      dynamicActivate('en');
    });
    useFetchCollaboratorsMock.mockReturnValue({
      collaborators: mockCollaboratorsData,
      getAllCollaborators: jest.fn(),
      hasCollaboratorsFailed: false,
      wasCollaboratorsFetched: false,
      isFetchingCollaborators: false
    });
    useUpdateCollaboratorMock.mockReturnValue({
      isUpdating: false,
      wasCollaboratorUpdated: false,
      updateCollaborator: mockUpdateCollaborator,
      hasCollaboratorFailed: false,
      errorMessage: '',
      reset(): void {
        throw new Error('Function not implemented.');
      }
    });
  });

  it('should render', () => {
    const { container } = renderComponent();
    expect(container).toMatchSnapshot();
  });

  it('should render the modal', () => {
    renderComponent();
    expect(screen.getByTestId('update-collaborator-modal')).toBeInTheDocument();
  });

  it('useCollaborators should be called', () => {
    renderComponent();
    expect(useFetchCollaboratorsMock).toHaveBeenCalled();
  });

  it('useUpdateCollaborator should be called', () => {
    renderComponent();
    expect(useUpdateCollaboratorMock).toHaveBeenCalled();
  });

  it('should show collaborator email in the modal', () => {
    renderComponent();
    expect(screen.getByTestId('collaborator-email').textContent).toBe(
      collaboratorMockValue.data[0].email
    );
  });

  it('should show collaborator name in the modal', () => {
    renderComponent();
    expect(screen.getByTestId('collaborator-name').textContent).toBe(
      collaboratorMockValue.data[0].name
    );
  });

  // collaborator without update:collaborator permission should not be able to update collaborator
  it('should disable edit button if user does not have update:collaborator permission', () => {
    // mock shouldDisableDeleteButton to return true
    const mockShouldDisableDeleteButton = jest.fn().mockReturnValue(true);
    jest.mock('components/Collaborators/useSafeDisableButton', () => ({
      ...jest.requireActual('hooks/useSafeDisableButton'),
      useSafeDisableButton: () => ({
        isDisabled: mockShouldDisableDeleteButton()
      })
    }));
    renderComponent();
    const editButton = screen.getByTestId('collaborator-button');
    fireEvent.click(editButton);
    expect(editButton).toBeDisabled();
  });

  // it('should call deleteHandler function when delete button is clicked', () => {
  //   renderComponent();
  //   userEvent.click(screen.getByTestId('delete-collaborator-button'));

  //   expect(mockDeleteCollaborator).toHaveBeenCalled();
  // });

  afterAll(() => {
    jest.clearAllMocks();
  });
});
