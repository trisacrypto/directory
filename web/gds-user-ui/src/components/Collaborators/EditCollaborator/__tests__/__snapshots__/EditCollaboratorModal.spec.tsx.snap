// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpdateCollaboratorModal should render 1`] = `
.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  position: relative;
  white-space: nowrap;
  vertical-align: middle;
  outline: 2px solid transparent;
  outline-offset: 2px;
  width: auto;
  line-height: 1.2;
  border-radius: var(--chakra-radii-md);
  font-weight: var(--chakra-fontWeights-semibold);
  transition-property: var(--chakra-transition-property-common);
  transition-duration: var(--chakra-transition-duration-normal);
  height: var(--chakra-sizes-10);
  min-width: var(--chakra-sizes-10);
  font-size: var(--chakra-fontSizes-md);
  -webkit-padding-start: var(--chakra-space-4);
  padding-inline-start: var(--chakra-space-4);
  -webkit-padding-end: var(--chakra-space-4);
  padding-inline-end: var(--chakra-space-4);
  background: var(--chakra-colors-transparent);
  color: blue;
}

.emotion-0:focus,
.emotion-0[data-focus] {
  background: var(--chakra-colors-transparent);
}

.emotion-0[disabled],
.emotion-0[aria-disabled=true],
.emotion-0[data-disabled] {
  opacity: 0.4;
  cursor: not-allowed;
  box-shadow: var(--chakra-shadows-none);
}

.emotion-0:hover,
.emotion-0[data-hover] {
  background: var(--chakra-colors-transparent);
}

.emotion-0:active,
.emotion-0[data-active] {
  background: var(--chakra-colors-gray-300);
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-top: var(--chakra-space-2);
}

.emotion-2>*:not(style)~*:not(style) {
  margin-top: var(--chakra-space-4);
  -webkit-margin-end: 0px;
  margin-inline-end: 0px;
  margin-bottom: 0px;
  -webkit-margin-start: 0px;
  margin-inline-start: 0px;
}

.emotion-4 {
  font-weight: 700;
}

.emotion-5 {
  text-transform: capitalize;
}

.emotion-7 {
  width: 100%;
  position: relative;
}

.emotion-8 {
  display: block;
  text-align: start;
  font-size: var(--chakra-fontSizes-md);
  -webkit-margin-end: var(--chakra-space-3);
  margin-inline-end: var(--chakra-space-3);
  margin-bottom: var(--chakra-space-2);
  font-weight: var(--chakra-fontWeights-medium);
  transition-property: var(--chakra-transition-property-common);
  transition-duration: var(--chakra-transition-duration-normal);
  opacity: 1;
}

.emotion-8[disabled],
.emotion-8[aria-disabled=true],
.emotion-8[data-disabled] {
  opacity: 0.4;
}

.emotion-10 {
  position: relative;
  pointer-events: auto;
}

.emotion-11 {
  z-index: 9999;
  border: 0;
  clip: rect(1px, 1px, 1px, 1px);
  height: 1px;
  width: 1px;
  position: absolute;
  overflow: hidden;
  padding: 0;
  white-space: nowrap;
}

.emotion-13 {
  width: 100%;
  min-width: 0px;
  outline: 2px solid transparent;
  outline-offset: 2px;
  position: relative;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  transition-property: var(--chakra-transition-property-common);
  transition-duration: var(--chakra-transition-duration-normal);
  font-size: var(--chakra-fontSizes-md);
  -webkit-padding-start: var(--chakra-space-4);
  padding-inline-start: var(--chakra-space-4);
  -webkit-padding-end: var(--chakra-space-4);
  padding-inline-end: var(--chakra-space-4);
  height: auto;
  border-radius: 0px;
  border: 1px solid;
  border-color: inherit;
  background: #E3EBEF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0px;
  overflow: hidden;
  min-height: var(--chakra-sizes-10);
}

.emotion-13:hover,
.emotion-13[data-hover] {
  border-color: var(--chakra-colors-gray-300);
}

.emotion-13[aria-readonly=true],
.emotion-13[readonly],
.emotion-13[data-readonly] {
  box-shadow: none!important;
  -webkit-user-select: all;
  -moz-user-select: all;
  -ms-user-select: all;
  user-select: all;
}

.emotion-13[disabled],
.emotion-13[aria-disabled=true],
.emotion-13[data-disabled] {
  opacity: 0.4;
  cursor: not-allowed;
}

.emotion-13[aria-invalid=true],
.emotion-13[data-invalid] {
  border-color: #E53E3E;
  box-shadow: 0 0 0 1px #E53E3E;
}

.emotion-13:focus,
.emotion-13[data-focus] {
  z-index: 1;
  border-color: #3182ce;
  box-shadow: 0 0 0 1px #3182ce;
}

.emotion-14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding: 0.125rem 1rem;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-overflow-scrolling: touch;
  position: relative;
  overflow: hidden;
}

.emotion-15 {
  color: var(--chakra-colors-gray-400);
  -webkit-margin-start: 0.125rem;
  margin-inline-start: 0.125rem;
  -webkit-margin-end: 0.125rem;
  margin-inline-end: 0.125rem;
  position: absolute;
  top: 50%;
  -webkit-transform: translateY(-50%);
  -moz-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.emotion-16 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  display: inline-grid;
  grid-area: 1/1/2/3;
  grid-template-columns: 0 min-content;
  color: inherit;
}

.emotion-16::after {
  content: attr(data-value) " ";
  visibility: hidden;
  white-space: pre;
  padding: 0px;
  grid-area: 1/2;
  font: inherit;
  min-width: 2px;
  border: 0;
  margin: 0px;
  outline: 2px solid transparent;
  outline-offset: 2px;
}

.emotion-17 {
  color: inherit;
  background: 0;
  opacity: 1;
  width: 100%;
  padding-top: 0.125rem;
  padding-bottom: 0.125rem;
  grid-area: 1/2;
  font: inherit;
  min-width: 2px;
  border: 0;
  margin: 0px;
  outline: 2px solid transparent;
  outline-offset: 2px;
}

.emotion-18 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.emotion-19 {
  opacity: 1;
  border: 0;
  border-color: inherit;
  border-style: solid;
  border-left-width: 1px;
  height: 100%;
}

.emotion-20 {
  font-size: 20px;
  -webkit-padding-start: var(--chakra-space-4);
  padding-inline-start: var(--chakra-space-4);
  -webkit-padding-end: var(--chakra-space-4);
  padding-inline-end: var(--chakra-space-4);
  height: 100%;
  border-radius: 0px;
  border: 1px solid;
  border-color: inherit;
  background: var(--chakra-colors-gray-100);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  border-width: 0;
  cursor: pointer;
}

.emotion-21 {
  width: 1em;
  height: 1em;
  display: inline-block;
  line-height: 1em;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  color: currentColor;
  vertical-align: middle;
}

.emotion-22 {
  margin-top: var(--chakra-space-2);
  color: var(--chakra-colors-gray-500);
  line-height: var(--chakra-lineHeights-normal);
  font-size: var(--chakra-fontSizes-sm);
}

.emotion-23 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  position: relative;
  white-space: nowrap;
  vertical-align: middle;
  outline: 2px solid transparent;
  outline-offset: 2px;
  width: auto;
  line-height: 1.2;
  border-radius: var(--chakra-radii-md);
  font-weight: var(--chakra-fontWeights-semibold);
  transition-property: var(--chakra-transition-property-common);
  transition-duration: var(--chakra-transition-duration-normal);
  height: var(--chakra-sizes-10);
  min-width: 150px;
  font-size: var(--chakra-fontSizes-md);
  -webkit-padding-start: var(--chakra-space-4);
  padding-inline-start: var(--chakra-space-4);
  -webkit-padding-end: var(--chakra-space-4);
  padding-inline-end: var(--chakra-space-4);
  background: orange;
}

.emotion-23:focus,
.emotion-23[data-focus] {
  box-shadow: var(--chakra-shadows-outline);
}

.emotion-23[disabled],
.emotion-23[aria-disabled=true],
.emotion-23[data-disabled] {
  opacity: 0.4;
  cursor: not-allowed;
  box-shadow: var(--chakra-shadows-none);
}

.emotion-23:hover,
.emotion-23[data-hover] {
  background: orange;
}

.emotion-23:active,
.emotion-23[data-active] {
  background: var(--chakra-colors-gray-300);
}

.emotion-24 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  position: relative;
  white-space: nowrap;
  vertical-align: middle;
  outline: 2px solid transparent;
  outline-offset: 2px;
  width: auto;
  line-height: 1.2;
  border-radius: var(--chakra-radii-md);
  font-weight: var(--chakra-fontWeights-semibold);
  transition-property: var(--chakra-transition-property-common);
  transition-duration: var(--chakra-transition-duration-normal);
  height: var(--chakra-sizes-10);
  min-width: 150px;
  font-size: var(--chakra-fontSizes-md);
  -webkit-padding-start: var(--chakra-space-4);
  padding-inline-start: var(--chakra-space-4);
  -webkit-padding-end: var(--chakra-space-4);
  padding-inline-end: var(--chakra-space-4);
  color: link;
}

.emotion-24:focus,
.emotion-24[data-focus] {
  box-shadow: var(--chakra-shadows-outline);
}

.emotion-24[disabled],
.emotion-24[aria-disabled=true],
.emotion-24[data-disabled] {
  opacity: 0.4;
  cursor: not-allowed;
  box-shadow: var(--chakra-shadows-none);
}

.emotion-24:hover,
.emotion-24[data-hover] {
  background: var(--chakra-colors-gray-100);
}

.emotion-24:hover[disabled],
.emotion-24[data-hover][disabled],
.emotion-24:hover[aria-disabled=true],
.emotion-24[data-hover][aria-disabled=true],
.emotion-24:hover[data-disabled],
.emotion-24[data-hover][data-disabled] {
  background: initial;
}

.emotion-24:active,
.emotion-24[data-active] {
  background: var(--chakra-colors-gray-200);
}

<div>
  <button
    class="chakra-button emotion-0"
    data-testid="edit-collaborator-button"
    disabled=""
    type="button"
  >
    <svg
      fill="none"
      font-size="24px"
      height="1em"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M11 4H4a2 2 0 0 0-2 2v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2v-7"
      />
      <path
        d="M18.5 2.5a2.121 2.121 0 0 1 3 3L12 15l-4 1 1-4 9.5-9.5z"
      />
    </svg>
    <div
      data-testid="modal"
    >
      <div
        data-testid="overlay"
      />
      <div
        data-testid="content"
      >
        <div
          data-testid="update-collaborator-modal"
        >
          Edit collaborator Role
        </div>
        <div
          data-testid="close"
        />
        <div
          data-testid="body"
        >
          <p
            class="chakra-text emotion-1"
          >
            Select the collaborator’s role and save.
          </p>
          <div
            class="chakra-stack emotion-2"
          >
            <div
              class="emotion-1"
            >
              <p
                class="chakra-text emotion-4"
              >
                Collaborator Name & Email
              </p>
              <p
                class="chakra-text emotion-5"
                data-testid="collaborator-name"
              >
                test
              </p>
              <p
                class="chakra-text emotion-1"
                data-testid="collaborator-email"
              >
                test@local.dev
              </p>
            </div>
            <div
              class="chakra-form-control emotion-7"
              role="group"
            >
              <label
                class="chakra-form__label emotion-8"
                for="role"
                id="field-1-label"
              >
                <span
                  class="emotion-4"
                >
                  Change Role
                </span>
              </label>
              <div
                class=" emotion-10"
                id="role"
              >
                <span
                  class="emotion-11"
                  id="react-select-2-live-region"
                />
                <span
                  aria-atomic="false"
                  aria-live="polite"
                  aria-relevant="additions text"
                  class="emotion-11"
                />
                <div
                  class=" emotion-13"
                >
                  <div
                    class=" emotion-14"
                  >
                    <div
                      class=" emotion-15"
                      id="react-select-2-placeholder"
                    >
                      Select Role
                    </div>
                    <div
                      class=" emotion-16"
                      data-value=""
                    >
                      <input
                        aria-autocomplete="list"
                        aria-controls="react-select-2-listbox"
                        aria-describedby="react-select-2-placeholder"
                        aria-expanded="false"
                        aria-haspopup="true"
                        aria-owns="react-select-2-listbox"
                        autocapitalize="none"
                        autocomplete="off"
                        autocorrect="off"
                        class=" emotion-17"
                        id="field-1"
                        role="combobox"
                        spellcheck="false"
                        tabindex="0"
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                  <div
                    class=" emotion-18"
                  >
                    <hr
                      aria-orientation="vertical"
                      class="chakra-divider emotion-19"
                    />
                    <div
                      aria-hidden="true"
                      class=" emotion-20"
                    >
                      <svg
                        class="chakra-icon emotion-21"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                          fill="currentColor"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
                <input
                  name="role"
                  type="hidden"
                  value=""
                />
              </div>
              <div
                class="chakra-form__helper-text emotion-22"
                id="field-1-helptext"
              />
            </div>
          </div>
        </div>
        <div
          data-testid="footer"
        >
          <button
            class="chakra-button emotion-23"
            data-testid="update-collaborator-button"
            type="button"
          >
            Save
          </button>
          <button
            class="chakra-button emotion-24"
            type="button"
          >
            Close
          </button>
        </div>
      </div>
    </div>
  </button>
</div>
`;
